src/main/java/org/osm2world/console/Output.java

```diff
@@ -58,6 +59,11 @@ public static void output(Configuration config,
 			CLIArgumentsGroup argumentsGroup)
 		throws IOException {
 
+		if (config instanceof PropertiesConfiguration) {
+			config.addProperty("configPath",
+				new File(((PropertiesConfiguration)config).getPath()).getParent());
```

Comment:
This location in the code is not necessarily reached – `ConversionFacade` should be able to be used directly, not just from `Output`. Not entirely sure what to do about this – perhaps you could at least additionally do an instanceof check directly in `resolveFileConfigProperty`? And move this `addProperty` call to `OSM2World.loadConfigFiles`?

Reply:
Moving it to `loadConfigFiles` is an improvement. However, `resolveFileConfigProperty` should still include code such as
```
if (basePath == null && config instanceof PropertiesConfiguration propertiesConfig
		&& propertiesConfig.getFile() != null) {
	basePath = propertiesConfig.getFile().getParent();
}
```
This is because `ConversionFacade` is intended as a public API, and when it is used in this way, the config passed to it may not have been obtained from `loadConfigFiles`.